lib/devos/mkFlake.nix

```diff
@@ -0,0 +1,198 @@
+{ lib, flake-utils, deploy }:
+let
+  mkFlake = { self, ... } @ args:
+      # TODO: pipe args through importers
+    mkFlake' args;
+
+  mkFlake' =
+    {
+      self
+      , name
+      , nixos
+      , overlay
+      , preOverlays ? [ ]
+      , systems ? [ "x86_64-linux" ]
+      , shell ? null
+      , config ? { allowUnfree = true; }
+
+      # internal & external module declarations
+      , nixosModules ? [ ]
+      , extraNixosModules ? [ ]
+      , extraSpecialArgs ? { }
+      , homeModules ? [ ]
+      , extraHomeModules ? [ ]
+      , extraHomeSpecialArgs ? { }
+      # , devShellModules
+
+      # applied devos environment configuriguration
+      , profiles
+      , users
+      , userProfiles
+      , suites
+      , hosts
+
+      # backports from alternate channel
+      , nixos' ? null
+      , backportPkgs ? pkgs: final: prev: { }
+      , backportModules ? [ ]
+      , disabledModules ? [ ]
+    }:
+    let
+
+      preOverlays_ =
+        preOverlays
+        ++
+        [
+          final: prev: let
+            system = prev.stdenv.hostPlatform;
+            # system = prev.stdenv.buildPlatform;
+          in
+            {
+              # add checks by conforming to simpleFlake checks api
+              "{$name}".checks =
+                let
+
+                  tests =
+                    lib.optionalAttrs
+                      (system  == "x86_64-linux")
+                      (
+                        import "${self}/tests" { inherit self; pkgs = prev; }
+                      );
+
+
+                  deployChecks =
+                    deploy.lib.${system}.deployChecks
+                      {
+                        nodes = let
+                          p = n: _:
+                            self.nixosConfigurations.${n}.config.nixpkgs.system == system;
+                        in lib.filterAttrs p self.deploy.nodes;
+                      };
+
+                in tests // deployChecks;
+
+              # add backports from nixos'
+              "{$name}".backports = # TODO: pros & cons of namespacing with `backports`
+                let
+
+                  nixosPkgs' = import nixos' {
+                    inherit system config;
+                    overlay = [ ];
+                  };
+
+                in backportPkgs nixosPkgs' final prev
+            }
+
+        ]
+
+      # produces `packages`, `legacyPackages` & `devShell` - but also a multi pkgs
+      outputs' = flake-utils.lib.simpleFlake' { # TODO: re-export pkgs in fork
+        inherit
+          self
+          name
+          systems
+          overlay
+          shell
+          config
+          ;
+        preOverlays = preOverlays_;
+        nixpkgs = nixos;
+      };
+      outputs = outputs'.flake;
+      pkgs = outputs'.pkgs;
```

Comment:
In order to be able to do this, I'll be making a patch to flake-utils, now. If we can't upstream, we probably can have that fork within `divnix`. Just in case we end up agreeing on this design.